### DISCUSSION TITLE
Pull Json data in flutter

### DISCUSSION THREAD

**Discussion by MrSmith100** [supabase/supabase#15435] [Questions]
Pulling from supabase, how do I get the json data. I have a table it has 2 columns called key, value

I want to pull back the json data in 'value' based on the value in 'key'

**Comment by GaryAustin1** [comment]
You probably should show an example of your table column data and what result you are trying to achieve.  The term "pull" itself is sort of unclear (or maybe that is a flutter specific term).

**Reply by MrSmith100** [reply]
Sorry. ok, so basically image one shows my database which shows 2 columns, 1 = key, 2 = value, and i want to pull that information into my flutter app. logic would be select value from table key_value_store where key = food
<img width="664" alt="image1" src="https://github.com/supabase/supabase/assets/53196521/3b3234c2-93d4-43ca-bb53-d97a6efa41a9">
 and my flutter code is: Future<void> _getReceivedData() async {
    await client
        .from("key_value_store")
        .select()
        .eq('key', 'food')
        .execute()
        .then((value) {
      if (value.error == null) {
        print('value.data: ${value.data}');
      } else {
        print(value.error.message.toString());
        print(value.error.hint.toString());
      }
    });
  }
  }

**Reply by GaryAustin1** [reply]
I don't know flutter so not commenting on syntax.   Normally data comes back in an array, even if one row.  You should get json back in data[0] in the case you show.  You can also use .single() to return a single value instead of an array if the result of your query is exactly one row.
